internal/cover/main.go

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// cover is a tool that runs `go test` with cross-package coverage on this
+// repository, ignoring any packages that opt out of coverage with .nocover
+// files. The coverage is written to a coverage.txt file in the current
+// directory.
+//
+// Usage
+//
+// Call cover with a list of one or more import paths of packages being
+// tested.
+//
+//   cover PKG ...
+//
+// This must be run from the root of the project.
+package main
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"go/build"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+func main() {
+	log.SetFlags(0)
+	if err := run(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+var (
+	errUsage        = errors.New("usage: cover packages")
+	errNoGoPackage  = errors.New("could not find a Go package in the current directory")
+	errNoImportPath = fmt.Errorf("could not determine import path for the Go package in the current directory")
+)
+
+func run() error {
+	packages := os.Args[1:]
```

Comment:
could take `packages` as an argument, reducing the interactions with globals in this function

Reply:
The pattern being followed here is just to avoid `log.Fatal` in
multiple places by having an alternative `main` which returns an error,
but sure I'll change it to accept the command line arguments as
parameters.